lib/datadog/lambda/utils/extension.rb

```diff
@@ -40,9 +42,10 @@ def self.check_extension_running
       File.exist?(EXTENSION_PATH)
     end
 
-    def self.send_start_invocation_request(event:)
+    def self.send_start_invocation_request(event:, request_context:)
       return unless extension_running?
 
+      request_headers[:LAMBDA_RUNTIME_AWS_REQUEST_HEADER_ID] = request_context.aws_request_id
       response = Net::HTTP.post(START_INVOCATION_URI, event.to_json, request_headers)
```

Comment:
You'll definitely have to test this. I'm worried that each time you reference `request_headers` that it makes the call to the `self.request_headers` method, and thus returns a new map object each time.